source/fundamentals/write-data.txt

```diff
@@ -23,20 +23,34 @@ Overview
 {+framework-core+} allows you to work with data in your application without
 explicitly running database commands. You can insert, update, or delete data
 within your application and persist those changes to MongoDB by using the
-``SaveChanges()`` method.
+``SaveChanges()`` or ``SaveChangesAsync()`` method.
 
-When you call the ``SaveChanges()`` method, the {+provider-short+} automatically detects
+When you call the ``SaveChanges()`` or ``SaveChangesAsync()`` method, the {+provider-short+} automatically detects
 any changes made to your data and runs the necessary commands to update the
 database by using the MongoDB Query API.
 
-In this guide, you can see examples of how to perform common write operations on
-an application configured to use the {+provider-short+}.
+The ``SaveChanges()`` and ``SaveChangesAsync()`` methods are transactional by
+default. This means that if an error occurs during an operation, the provider
+rolls back any changes made to the database. Because of this, your application
+must be connected to a transaction-capable deployment of MongoDB server, such as
+a replica set.
+
+You can disable transactions during write operations by setting
```

Comment:
I'd probably clarify this "You can disable these automatic transactions within SaveChanges and SaveChangeaAsync by setting..." as other transaction facilities in EF exist which we don't yet support (explicit transactions).

We should clarify that this is not recommended as any kind of concurrency change, database operation failure, duplicate key etc. will leave the database with only some of the changes applied.

Reply:
This was getting a little long and involved so I moved it to it's own section just below the overview and addressed these points there.